devcontainer.sh

```diff
@@ -73,7 +144,35 @@ MOUNT="${MOUNT} --mount type=bind,source=${WORKSPACE},target=${WORK_DIR}"
 debug "MOUNT: ${MOUNT}"
 
 echo "Building and starting container"
-DOCKER_IMAGE_HASH=$(docker build -f $DOCKER_FILE $ARGS .)
-debug "DOCKER_IMAGE_HASH: ${DOCKER_IMAGE_HASH}"
 
-docker run -it $REMOTE_USER $PORTS $ENVS $MOUNT -w $WORK_DIR $DOCKER_IMAGE_HASH $SHELL
\ No newline at end of file
+DOCKER_TAG=$(echo "$DOCKER_FILE" | md5sum - | awk '{ print $1 }')
+# shellcheck disable=SC2086
+docker build -f "$DOCKER_FILE" -t "$DOCKER_TAG" $ARGS .
+build_status=$?
+
+if [[ $build_status -ne 0 ]]; then
+    echo "Building docker image failed..." >&2
+    exit 7
+fi
+
+debug "DOCKER_TAG: ${DOCKER_TAG}"
+
+PUID=$(id -u)
+PGID=$(id -g)
+
+# shellcheck disable=SC2086
+docker run -it $DOCKER_OPTS $PORTS $ENVS $MOUNT -w "$WORK_DIR" "$DOCKER_TAG" "$SHELL" -c "\
+if [[ '$REMOTE_USER' != '' ]] && command -v usermod &>/dev/null; \
+then \
+    sudo=''
+    if [[ "$(stat -f -c '%u' $(which sudo))" = '0' ]]; then
```

Comment:
For some reason the mcr.microsoft.com/vscode/devcontainers/base:alpine-3.12 image has `sudo` owned by `1000` and therefore it cannot be used as root